packages/switch/src/react/index.tsx

```diff
@@ -0,0 +1,192 @@
+import { compose, css } from 'glamor'
+import React, { HTMLAttributes } from 'react'
+
+import Halo from '@pluralsight/ps-design-system-halo'
+import {
+  useTheme,
+  names as themeNames
+} from '@pluralsight/ps-design-system-theme'
+import {
+  ValueOf,
+  isFunction,
+  RefForwardingComponent
+} from '@pluralsight/ps-design-system-util'
+
+import stylesheet from '../css'
+import * as vars from '../vars'
+
+const styles = {
+  switch: (disabled: boolean, labelAlign: ValueOf<typeof vars.labelAligns>) =>
+    compose(
+      css(stylesheet['.psds-switch']),
+      css(stylesheet[`.psds-switch--labelAlign-${labelAlign}`]),
+      disabled && css(stylesheet['.psds-switch--disabled'])
+    ),
+
+  track: (props: {
```

Comment:
I'm fine with not passing props to all these.  But there are functions with 3-4 params, and redefined SwitchProps.  I'd consider not defining these props in multiple places for this many.  

Something like:

```
(props: Pick<SwitchProps, 'checked', 'color', 'size'>) => ...
```

Seems DRYer.

Reply:
Look like that won't work